### PR TITLE
Ancillary

### PR DESCRIPTION
**Pull request by DanielJMaher**
Must go in after #234 because of https://github.com/DanielJMaher/compliance-checker/blob/cdl/compliance_checker/cf/util.py#L222

### CODE REVIEW

**Comment by DanielJMaher** [comment]
@lukecampbell

File: compliance_checker/cf/cf.py [hunk 1]
```diff
@@ -723,9 +725,9 @@ def check_ancillary_data(self, ds):
             anc_result = Result(BaseCheck.HIGH, name=('§3.4 Ancillary Variables', k))
             msgs = []
 
-            if not isinstance(anc, str):
+            if not isinstance(anc, str) and not isinstance(anc, unicode):
```
Comment (lukecampbell) [inline compliance_checker/cf/cf.py]: @daf @kwilcox isn't there some special python3 magic that has to be done here?

Comment (kwilcox) [inline compliance_checker/cf/cf.py]: This is fairly common (without the use of `six`) to support both py2 and py3:

```
try:
    basestring
except NameError:
    basestring = str
```

Then just always do `isinstance(foo, basestring)`

Comment (lukecampbell) [inline compliance_checker/cf/cf.py]: Thanks, I thought so. I don't do a lot of py2 and py3 support together.
@DanielJMaher would you please?

Comment (DanielJMaher) [inline compliance_checker/cf/cf.py]: @lukecampbell Im on it

Comment (kwilcox) [inline compliance_checker/cf/cf.py]: you can add that test in a utils function and always import it from there... i know its used other places in CC already.

File: compliance_checker/cf/cf.py [hunk 2]
```diff
@@ -723,9 +725,9 @@ def check_ancillary_data(self, ds):
...
                 anc_result.value = False
-                anc_result.msgs = ["ancillary_variables is not a string"]
+                anc_result.msgs = ["ancillary_variable(s) %s is not a string or unicode" % anc]
```
Comment (kwilcox) [inline compliance_checker/cf/cf.py]: You should use new style string formatting... this style is deprecated.

Comment (lukecampbell) [inline compliance_checker/cf/cf.py]: agreed.

File: compliance_checker/cf/cf.py (hunk 1 above)
Comment (DanielJMaher) [inline compliance_checker/cf/cf.py]: https://github.com/DanielJMaher/compliance-checker/blob/ancillary/compliance_checker/cf/cf.py#L12-L15

Already there,

**Comment by DanielJMaher** [comment]
@lukecampbell @kwilcox

File: compliance_checker/cf/cf.py [hunk 3]
```diff
@@ -723,9 +725,9 @@ def check_ancillary_data(self, ds):
             anc_result = Result(BaseCheck.HIGH, name=('§3.4 Ancillary Variables', k))
             msgs = []
 
-            if not isinstance(anc, str):
+            if not isinstance(anc, basestring):
```
Comment (lukecampbell) [inline compliance_checker/cf/cf.py]: I think basestring is invalid in py3

Comment (daf) [inline compliance_checker/cf/cf.py]: https://github.com/DanielJMaher/compliance-checker/blob/ancillary/compliance_checker/cf/cf.py#L12-L15 defines it

Comment (DanielJMaher) [inline compliance_checker/cf/cf.py]: If you look at https://github.com/DanielJMaher/compliance-checker/blob/ancillary/compliance_checker/cf/cf.py#L12-L15 it catches if it is invalid and then defaults it to str  

This is per Kyle's guidance.  It should be fine in Py3

**Pull request merged**